pkg/build/builder/dockerutil.go

```diff
@@ -156,15 +156,27 @@ func dockerRun(client DockerClient, createOpts docker.CreateContainerOptions, lo
 
 	containerName := containerNameOrID(c)
 
-	// Container was created, so we defer its removal.
-	defer func() {
+	// Container was created, so we defer its removal, and also remove it if we get a sigterm
```

Comment:
That function appears to leak go routines, it has no handling for the case where the signal never arrives.  (calling Close() does not appear to do anything to terminate the go routine that I see), so it doesn't seem well suited for this use case.